File: 2018/6/part2.rb
points = File.readlines('input').map {|line| line.strip.split(', ').map{|value| value.to_i}}
# puts points

def manhattan_distance(x,y, points)
    points.collect{|point| (x - point[0]).abs + (y - point[1]).abs}
end

max_x = points.collect {|x| x[0] }.max
max_y = points.collect {|x| x[1] }.max

size = 0
for x in 0..max_x
    for y in 0..max_y
        distances = manhattan_distance(x,y, points)
        if distances.sum <10000
            size = size +1
        end
    end
end

p size

File: 2024/1/part2.py
first_list = []
second_list = []
with open('input') as f:
    lines = f.readlines()
    for line in lines:
        a,b = line.split()
        first_list.append(int(a))
        second_list.append(int(b))

sum = 0
for a in first_list:
    print(a*second_list.count(a))
    sum+=(a*second_list.count(a))
print(sum)